iib/workers/tasks/opm_operations.py

```diff
@@ -738,3 +751,110 @@ def opm_create_empty_fbc(
         binary_image=binary_image,
         dockerfile_name='index.Dockerfile',
     )
+
+
+def opm_registry_add_fbc_fragment(
+    temp_dir: str,
+    from_index: str,
+    binary_image: str,
+    fbc_fragment: str,
+    overwrite_from_index_token: Optional[str],
+) -> None:
+    """
+    Add FBC fragment to from_index image.
+
+    This only produces the index.Dockerfile file and does not build the container image.
+
+    :param str temp_dir: the base directory to generate the database and index.Dockerfile in.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from. This should point to a digest or stable tag.
+    :param str fbc_fragment: the fbc fragment to be added in from_index.
+    :pararm str overwrite_from_index_token: token used to access the image
+    """
+    fragment_path, fragment_operator = extract_fbc_fragment(temp_dir, fbc_fragment)
+
+    is_operator_in_db, index_db_path = verify_operator_exists(
+        from_index=from_index,
+        base_dir=temp_dir,
+        operator_package=fragment_operator,
+        overwrite_from_index_token=overwrite_from_index_token,
+    )
+
+    from_index_configs_dir = get_catalog_dir(from_index=from_index, base_dir=temp_dir)
+
+    if is_operator_in_db:
+        log.info('Removing %s from %s index.db ', fragment_operator, from_index)
+        _opm_registry_rm(index_db_path, [fragment_operator], temp_dir)
+        migated_catalog_dir, _ = opm_migrate(
+            index_db=index_db_path,
+            base_dir=temp_dir,
+            generate_cache=False,
+        )
+
+        for operator_package in os.listdir(migated_catalog_dir):
+            shutil.copytree(
+                os.path.join(migated_catalog_dir, operator_package),
+                os.path.join(from_index_configs_dir, operator_package),
+                dirs_exist_ok=True,
+            )
+
+    # copy fragment_operator to from_index configs
+    fragment_opr_dest_path = os.path.join(from_index_configs_dir, fragment_operator)
+    if os.path.exists(fragment_opr_dest_path):
+        shutil.rmtree(fragment_opr_dest_path)
+    shutil.copytree(os.path.join(fragment_path, fragment_operator), fragment_opr_dest_path)
+
+    # Generate cache as part of opm_generate_dockerfile()?
+    local_cache_path = os.path.join(temp_dir, 'cache')
+    generate_cache_locally(temp_dir, from_index_configs_dir, local_cache_path)
```

Comment:
This might fail if cache was already generated. Should we delete it if it exist first?